src/Evolution/Systems/Cce/Equations.cpp

```diff
@@ -269,4 +337,66 @@ void ComputeBondiIntegrand<Tags::LinearFactorForConjugate<Tags::BondiH>>::
                    j * (conj(j) * dy_j + j * conj(dy_j)) / (1.0 + j * conj(j)));
   *linear_factor_for_conjugate_h = j * (*script_djbar);
 }
+
+void ComputeBondiIntegrand<Tags::PoleOfIntegrand<Tags::KleinGordonPi>>::
+    apply_impl(gsl::not_null<SpinWeighted<ComplexDataVector, 0>*>
+                   pole_of_integrand_for_kg_pi,
+               const SpinWeighted<ComplexDataVector, 1>& eth_kg_psi,
+               const SpinWeighted<ComplexDataVector, 1>& bondi_u) {
+  *pole_of_integrand_for_kg_pi =
+      -real(bondi_u.data() * conj(eth_kg_psi).data());
+}
+
+void ComputeBondiIntegrand<Tags::RegularIntegrand<Tags::KleinGordonPi>>::
+    apply_impl(gsl::not_null<SpinWeighted<ComplexDataVector, 0>*>
+                   regular_integrand_for_kg_pi,
+               // pre_swsh_derivative_tags
+               const SpinWeighted<ComplexDataVector, 0>& dy_dy_kg_psi,
+               const SpinWeighted<ComplexDataVector, 0>& dy_kg_psi,
+               const SpinWeighted<ComplexDataVector, 1>& dy_bondi_u,
+               const SpinWeighted<ComplexDataVector, 0>& dy_w,
+               // swsh_derivative_tags
+               const SpinWeighted<ComplexDataVector, 1>& eth_dy_kg_psi,
+               const SpinWeighted<ComplexDataVector, 2>& eth_eth_kg_psi,
+               const SpinWeighted<ComplexDataVector, 1>& eth_kg_psi,
+               const SpinWeighted<ComplexDataVector, 1>& ethbar_j,
+               const SpinWeighted<ComplexDataVector, 0>& ethbar_u,
+               const SpinWeighted<ComplexDataVector, 1>& eth_beta,
+               const SpinWeighted<ComplexDataVector, 1>& eth_j_jbar,
+               const SpinWeighted<ComplexDataVector, 0>& eth_ethbar_kg_psi,
+               // integration_independent_tags
+               const SpinWeighted<ComplexDataVector, 2>& j,
+               const SpinWeighted<ComplexDataVector, 0>& exp2beta,
+               const SpinWeighted<ComplexDataVector, 0>& du_r_divided_by_r,
+               const SpinWeighted<ComplexDataVector, 0>& one_minus_y,
+               const SpinWeighted<ComplexDataVector, 1>& eth_r_divided_by_r,
+               const SpinWeighted<ComplexDataVector, 0>& bondi_r,
+               const SpinWeighted<ComplexDataVector, 0>& bondi_k,
+               const SpinWeighted<ComplexDataVector, 1>& bondi_u,
+               const SpinWeighted<ComplexDataVector, 0>& bondi_w) {
+  SpinWeighted<ComplexDataVector, 1> eth_k = 0.5 * eth_j_jbar / bondi_k;
+  SpinWeighted<ComplexDataVector, 0> from_lhs =
+      du_r_divided_by_r * one_minus_y * dy_dy_kg_psi;
```

Comment:
Why is this "from_lhs"? It's on the RHS of the evolution eqn for $\Pi$?

Reply:
This is a bit technical. Since the radial grid ($\breve{y}$) is not the Bondi radius ($r$), i.e., see Eq. (44) in Jordan's [paper](https://arxiv.org/pdf/2007.01339.pdf), the actual solved variable is $\breve{\Pi}=\partial_u\Psi |_{y}$ rather than $\Pi=\partial_u\Psi |_r$, they are related through $\breve{\Pi}=\Pi+\partial_u R\partial_r\Psi$, see Eq. (47) in Jordan's [paper](https://arxiv.org/pdf/2007.01339.pdf). As a result, the hypersurface equation for $\breve{\Pi}$: $(1-y)\partial_y\breve{\Pi}+\breve{\Pi}=...$ gets an additional factor comparing with the one for $\Pi$. The new term arises from the left hand side: by replacing $\Pi$ with $\breve{\Pi}$.

More details will be provided in the upcoming method paper.

Reply:
Ahh ok makes sense. I'm not really familiar with this equations, but is this obvious from knowing what the quantities represent? If it's not, could you add a little note that this term comes from switching between $\breve\Pi$ to $\Pi$?

Reply:
I've added a note one line above. More mathematical details will be in the upcoming method paper. I will add a reference once the paper is out.